File: 20150915-StringCalculator-Java8/src/test/java/kata/SCTest.java
package kata;

import org.junit.Test;

import static kata.SC.add;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.fail;

public class SCTest {
    @Test
    public void given_a_blank_string_should_return_0() {
        assertEquals(0, add(""));
    }

    @Test
    public void given_a_number_should_return_its_value() {
        assertEquals(123, add("123"));
    }

    @Test
    public void given_numbers_separated_by_a_comma_or_newline_should_return_the_sum() {
        assertEquals(6, add("1,2\n3"));
    }

    @Test
    public void given_numbers_separated_by_a_custom_single_character_separator_should_return_the_sum_of_all_less_than_1001() {
        assertEquals(6, add("//*\n1*2*3*1003"));
    }

    @Test
    public void given_numbers_separated_by_a_custom_single_multi_character_separator_should_return_the_sum_of_all_less_than_1001() {
        assertEquals(6, add("//[***][===]\n1===2***3***1003"));
    }

    @Test
    public void given_numbers_with_at_least_one_negative_should_throw_an_exception_with_the_negatives_in_the_exception_message() {
        try {
            add("//*\n1*-2*3*-4");
            fail();
        } catch (IllegalArgumentException ex) {
            assertEquals("-2,-4", ex.getMessage());
        }
    }
}

File: 20160425-StringCalculator-Kotlin/src/test/kotlin/kata/SCTest.kt
package kata

import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.fail

class SCTest {
    @Test fun given_a_blank_string_should_return_0() {
        assertEquals(0, add(""))
    }

    @Test fun given_a_number_should_return_its_value() {
        assertEquals(5, add("5"))
    }

    @Test fun given_numbers_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("1,2,1001,3"))
    }

    @Test fun given_numbers_separated_with_comma_or_newline_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("1,2\n3"))
    }

    @Test fun given_numbers_separated_with_a_single_character_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("//;\n1;2001;2;23334;3"))
    }

    @Test fun given_numbers_separated_with_a_single_regex_character_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("//*\n1*2*30301*3"))
    }

    @Test fun given_numbers_with_at_least_one_negative_show_throw_an_exception_with_the_negatives_in_the_exception_message() {
        try {
            add("1,-2,3,-4")
            fail()
        } catch(e: IllegalArgumentException) {
            assertEquals("-2,-4", e.message)
        }
    }

    @Test fun given_number_separated_with_a_single_multi_character_separator_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("//[***]\n1***2001***2***23334***3"))
    }

    @Test fun given_number_separated_with_multiple_multi_character_separator_should_return_the_sum_of_all_numbers_less_than_1001() {
        assertEquals(6, add("//[***][===]\n1===2001***2===23334***3"))
    }
}
